autoscheduler/frontend/src/utils/throttle.ts

```diff
@@ -45,11 +45,12 @@ const throttle = ((): (id: string, callback: any, delay: number, runOtherIDs: bo
       });
     }
 
-    // Update function to run when interval completes
-    functions.set(id, callback);
-
     // Update beforeunload event listener so callback is fired before close/refresh
+    window.removeEventListener('beforeunload', functions.get(id));
     window.addEventListener('beforeunload', callback);
+
+    // Update function to run when interval completes
+    functions.set(id, callback);
```

Comment:
Mind explaining how this "fixes the incorrect removal of event listener in throttle?

Reply:
I was setting the new function before the `removeEventListener` call before (which needs you to use the same callback as the event listener was added with), so it wasn't actually removing it